src/js-form-validator.ts

```diff
@@ -313,10 +313,8 @@ export class FormValidator {
       case 'phone':
         if (!allowEmpty && !value.length) return -1;
         let vals = value.split('x');
-        for (let i = 0, l = vals.length; i < l; i++)
-          vals[i] = vals[i].replace(/\D/g, '');
         value = vals.join('x').replace(/^1/, ''); // remove leading 1 if added, only for US
-        status = /^[0-9x]{7,16}$/.test(value) ? 1 : 0;
+        status = /^[0-9\-\+\s\(\)]{7,16}$/.test(value) ? 1 : 0;
```

Comment:
@Izzmo if we only want to reference the source files would we only being using the compiler for the sake of error finding?

I think we are after the same goal with only being worried about the numbers. The issue that was happening was that since we were stripping out those other characters the entry was able to scatter letters in the entry and it would pass validation, which is what was reported as a bug during testing. Would another possibility be to have a regex before the stripping out of characters to block specific letters we don't want to allow, and then after taking everything out but numbers to run the second regex? I don't think we want to strongly restrict user entry for these forms, which makes what to allow and what not to difficult. I attempted to allow as many of the special characters that we may want to allow a user to have access to while blocking things like letters and the like. What would some of your ideas be on how we can block these letters making it through the validation?